Marlin/src/sd/cardreader.cpp

```diff
@@ -472,46 +472,48 @@ void CardReader::mount() {
 #endif
 
 void CardReader::manage_media() {
-  static uint8_t prev_stat = 2;     // First call, no prior state
-  uint8_t stat = uint8_t(IS_SD_INSERTED());
-  if (stat == prev_stat) return;
+  static struct { bool present:1, inited:1; } media_stat = { false, false };
 
-  DEBUG_SECTION(mm, "CardReader::manage_media", true);
-  DEBUG_ECHOLNPGM("SD Status ", prev_stat, " -> ", stat);
+  bool is_present = IS_SD_INSERTED();
+  if (is_present == media_stat.present) return; // No change in media presence?
```

Comment:
We do want to proceed during init even if there is no media because `ui.media_changed` might want to init the UI in tandem with `manage_media` to indicate no media. `MarlinUI::media_changed` is aware that `prev_stat` could be 2 and does some different handling in that case.